08-single-page-web-apps/mongo_mongoose/readme.md

```diff
@@ -480,129 +435,463 @@ kitty.save(function (err) {
 });
 ```
 
-## Exercise: YUM!  
-
-It's time to play with our restaurants.
+Ok well thats nice. But let's see how mongoose plays with express by building an app called [reminders](https://github.com/ga-dc/reminders_mongo)
 
-Refer to [Getting Started](http://mongoosejs.com/docs/index.html)
+## Reminders- Schema, Models & Seeds
 
-### Create a new js "app"
+Lets create a brand new express application and grab up all the dependencies we'll need
 
+```bash
+$ mkdir reminders_mongo
+$ cd reminders_mongo
+$ npm init
+$ npm install express --save
+$ npm install hbs --save
+$ npm install body-parser --save
+$ npm install method-override --save
+$ npm install mongoose --save
+$ touch app.js
 ```
-mkdir
-npm init
-npm install --save mongoose
+
+> The 5 dependencies we'll be using for this app are:
+  1. `express` - web Frameworks
+  2. `hbs` - view engine
+  3. `body-parser` - allows us to get parameter values from forms
+  4. `method-override` - allows us to do put/delete requests in our hbs views
+  5. `mongoose` - our Mongo ORM
+
+Let's first start by defining our schema, models and creating some seed data.
+
+Folders/files:
+
+```bash
+$ mkdir db
+$ mkdir models
+$ touch models/author.js
+$ touch models/reminder.js
+$ touch db/schema.js
+$ touch db/seeds.js
 ```
 
-### Create a Connection.  Verify it.
+Just like in `active-record` we will define the structure of our database using schemas
 
+Instead of defining tables we'll be defining documents.
+
+In `db/schema.js`:
+
+```js
+// requiring mongoose dependency
+var mongoose = require('mongoose')
+
+// instantiate a name space for our Schema constructor defined by mongoose.
+var Schema = mongoose.Schema,
+    ObjectId = Schema.ObjectId
+
+// defining schema for reminders
+var ReminderSchema = new Schema({
+  body: String
+})
+
+// defining schema for authors.
+var AuthorSchema = new Schema({
+  name: String,
+  reminders: [ReminderSchema]
+})
+
+// setting models in mongoose utilizing schemas defined above
+var AuthorModel = mongoose.model("Author", AuthorSchema)
+var ReminderModel = mongoose.model("Reminder", ReminderSchema)
 ```
-nodemon index.js
+
+> Note the syntax for reminders  in the `AuthorSchema`. This signifies it will be an array of reminder documents
+
+Next, let's define our models that will provide the interface for queries and other logic.
+
+In `models/author.js`:
+
+```js
+require("../db/schema")
+var mongoose = require('mongoose')
+
+var AuthorModel = mongoose.model("Author")
+module.exports = AuthorModel
 ```
 
-``` js
-var mongoose = require('mongoose');
-mongoose.connect('mongodb://localhost/milk-n-cookies');
+In `models/reminder.js`:
 
-var db = mongoose.connection;
-db.on('error', console.error.bind(console, 'connection error:'));
-db.once('open', function (callback) {
-  console.log("Connection established to: ", db.name)
-});
```

Comment:
You removed that message that indicates we have a connection?  Are we so focused on the application that we have lost sight of the mongoose lesson?  As I continue reviewing, I see an awful lot of code that has nothing to do with mongo/mongoose.  To be clear, I'm not expecting it to be changed tonight.  Let's just keep an eye on this for future updates.